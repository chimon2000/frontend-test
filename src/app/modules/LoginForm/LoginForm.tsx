import * as React from "react";

import Button from "../../components/Button";
import Input from "../../components/Input";

import { render } from "react-dom";

import { useLoginForm } from "./form.hooks";

export interface LoginFormProps {
  login: Function;
}

export const LoginForm: React.FunctionComponent<LoginFormProps> = ({
  login
}) => {
  const { values, handleInputChange, handleSubmit } = useLoginForm(
    { token: null },
    login
  );
  return (
    <div className="module-login-form">
      <form onSubmit={handleSubmit} className="login-form">
        <h2 className="login-form__header">"Login" with a Yelp API Key</h2>
        <p>
          Make sure you are enrolled in the{" "}
          <a
            target="__blank"
            href="https://www.yelp.com/developers/v3/manage_app"
          >
            Developer Beta
          </a>{" "}
          to access GraphQL.
        </p>
        <div>
          <Input
            type="text"
            name="token"
            label="API Key"
            id="input-token"
            block
            className="login-form__input input input--text"
            onChange={handleInputChange}
          />
          <Button type="submit" block className="login-form__submit">
            Submit
          </Button>
        </div>
      </form>
    </div>
  );
};
