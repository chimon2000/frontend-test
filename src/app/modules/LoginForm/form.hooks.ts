import * as React from "react";

export const useLoginForm = (
  initialValues: { [name: string]: any },
  callback: Function
) => {
  const [values, setValue] = React.useState(initialValues);

  const handleSubmit = (e: React.FormEvent) => {
    if (e) {
      e.preventDefault();
    }
    callback(values);
  };

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    e.persist();
    setValue((values: {}) => ({ ...values, [e.target.name]: e.target.value }));
  };

  return {
    handleSubmit,
    handleInputChange,
    values
  };
};
