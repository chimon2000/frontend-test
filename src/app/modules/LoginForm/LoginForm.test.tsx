import * as React from "react";
import { shallow } from "enzyme";

import { LoginForm } from "./LoginForm";

describe("<LoginForm />", () => {
  it("doesn't blow up", () => {
    const wrapper = shallow(<LoginForm login={() => {}} />);

    expect(wrapper).toMatchSnapshot();
  });
});
