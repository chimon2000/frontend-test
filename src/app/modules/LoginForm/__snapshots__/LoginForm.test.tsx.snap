// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> doesn't blow up 1`] = `
<div
  className="module__login-form"
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <h2
      className="login-form__header"
    >
      "Login" with a Yelp API Key
    </h2>
    <p>
      Make sure you are enrolled in the
       
      <a
        href="https://www.yelp.com/developers/v3/manage_app"
        target="__blank"
      >
        Developer Beta
      </a>
       
      to access GraphQL.
    </p>
    <div>
      <Component
        block={true}
        className="login-form__input input input--text"
        id="input-token"
        label="API Key"
        name="token"
        onChange={[Function]}
        type="text"
      />
      <Component
        block={true}
        className="login-form__submit"
        type="submit"
      >
        Submit
      </Component>
    </div>
  </form>
</div>
`;
